sklearn/tests/test_multiclass.py

```diff
@@ -921,3 +922,108 @@ def test_ovo_consistent_binary_classification():
     ovo.fit(X, y)
 
     assert_array_equal(clf.predict(X), ovo.predict(X))
+
+
+@pytest.mark.usefixtures("enable_slep006")
+def test_fitted_coefs_differ_when_sample_weight_differs():
+    """Assert if coefficients differ when estimators are fitted with different
+    `sample_weights`."""
+    sample_weight = np.ones_like(iris.target, dtype=np.float64)
+    multiclass_classifiers = [
+        OneVsRestClassifier,
+        OneVsOneClassifier,
+        OutputCodeClassifier,
+    ]
+    for classifier in multiclass_classifiers:
+        clf = classifier(
+            estimator=SGDClassifier(random_state=42).set_fit_request(sample_weight=True)
+        )
+        # fitting with sample_weight=sample_weight
+        clf.fit(iris.data, iris.target, sample_weight=sample_weight)
+        # TODO: should we raise an error if sample weight is requested but is not
+        # passed in the router's fit method when enable slep006 is True?
+        # clf.fit(iris.data, iris.target)
```

Comment:
@glemaitre see https://github.com/scikit-learn/scikit-learn/issues/23920

Reply:
OK I see, we can remove the comment then.

We would need to settle on the behaviour at some point. I am a bit on the same side as @thomasjpfan, finding that this is weird not to raise anything (I can be happy with a warning).